### DISCUSSION TITLE
tutorials/lift-off-part2/10-journeys-end

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#29] [course feedback]
# tutorials/lift-off-part2/10-journeys-end

https://www.apollographql.com/tutorials/lift-off-part2/10-journeys-end

**Comment by Hydrock** [comment]
Thank you so much for your hard work! Although there were problems in this course, I understand how difficult it is to create good lessons. You have one of the best sites and for free!

**Reply by neogeek13** [reply]
100% agree. I've been responsible for training content and this is a great set of training on a complex topic (and free!).

**Comment by neogeek13** [comment]
Thank you for these detailed, closely documented and versatile courses. (I really appreciate the support for TypeScript!) Supporting both visual and reading learners is a really helpful touch! <3

**Comment by sureshmenon87** [comment]
This course is precious!!  Structured very excellent!! I really enjoyed and learned a load of things.  Appreciate your hardwork on making this course. Thanks

**Comment by konstantinkrumin** [comment]
Thank you so much for this course!

It explains the concepts needed in easy to follow and thorough manner. I love how interactive this course is, with a right combination of videos, text and assignments all fitting into one page making it a seamless experience.

**Comment by nwazuo** [comment]
Amazing! One of the best course experiences I've had. I love how some of the exercises were intentionally designed to make sure you've done the hands-on practice by asking you questions you can only answer if you've really done them.

**Comment by Yasir5247** [comment]
Just one simple Question. How doest RESTDataSource solve N+1 problem. doest the RESTDataSource implement DataLoader package from facebook? or similar implementation

**Reply by mabuyo** [reply]
Hey there @Yasir5247 ! Good question. [In the lesson about Apollo `RESTDataSource`](https://www.apollographql.com/tutorials/lift-off-part2/03-apollo-restdatasource), we mention that the `RESTDataSource` class will efficiently handle resource caching and deduplication for our REST API calls. In our case, this means it will make sure that we're not sending the same `GET /author/1` HTTP request over and over in parallel.

The `DataLoader` package you're referring to is separate, and focuses on _batching_. [Here's a link to a doc](https://www.apollographql.com/docs/apollo-server/data/fetching-rest/#using-with-dataloader) that explains more about using `DataLoader` with `RESTDataSource`.

Hope that helps!

**Comment by vikyw89** [comment]
the amount of boilerplate is ungodly,
the so called type generator for TS is so complicated,

the concept starts out good, but the typegen part is really bad, not a good DX at all.

**Reply by lizhennessy** [reply]
Hi @vikyw89, we hear you! It's definitely a lot of extra work to ensure that type safety is in place for our code, as well as to follow the code-gen process! I am happy to revisit the TS-specific lessons to make sure that we're spending enough time covering the concepts _and_ the details with equal care and attention. Please don't hesitate to reach out if there's anything else you feel would lead to a more positive experience for future learners.

**Comment by le-el** [comment]
These course is splendiferous ever.
I will study other courses continuously.
Thanks for great course.😊

**Comment by repiemm** [comment]
Awesome!, I didn't experience any inconvenience in this course, the steps were clear and the exercises were simple enough to get the idea without too much hassle

**Reply by cdwheatley** [reply]
Yes, was great not to run in to any CAT-tastrophes.

**Comment by mianshah** [comment]
Two things.......
First, great tutorial put together it couldn't be more simpler and digestive then this.

Second: good God gracious! So much typing, overhead and side effects, just to send a request and get som JOSON. With all short comings REST api is way way more predictable and down to earth simpler to get the job don. 
The Apollo team really need to rework the over all architecture and make use type less and remember all those side effects keywords. For example, why cannot I create Apollo server with one simple method and a callback? Why I have to twist my arm to get thing running? Some of these steps could have been handled internally. The idea of Resolvers per field is an absurd idea. Why I cannot I pass you my schema and you return me an object of fields value based on my schema? Sometimes it's better to take a 10000 feet view from above of the architecture and think outside of of those fancy "patterns", as preached.......frameworks should always be KISS

**Comment by Ceheiss** [comment]
This sets a new standard for tutorials, I am really amazed how well structured it is. Kudos to everyone, really, you are able to break down this in an excellent manner
